src/openms/source/QC/DBSuitability.cpp

```diff
@@ -290,9 +229,320 @@ namespace OpenMS
     return true;
   }
 
-  bool DBSuitability::passesFDR_(const PeptideHit& hit, double FDR)
+  bool DBSuitability::passesFDR_(const PeptideHit& hit, double FDR) const
   {
     if (hit.getScore() > FDR) return false;
     return true;
   }
-}
+
+  pair<String, Param> DBSuitability::extractSearchAdapterInfoFromMetaValues_(const ProteinIdentification::SearchParameters& search_params) const
+  {
+    Param p;
+    // list of all allowed adapters
+    vector<String> working_adapters{ "CometAdapter", "CruxAdapter", "MSGFPlusAdapter", "MSFraggerAdapter", "MyriMatchAdapter", "OMSSAAdapter", "XTandemAdapter" };
+
+    String adapter;
+    vector<String> keys;
+    search_params.getKeys(keys);
+    for (const String& key : keys)
+    {
+      if (adapter.empty()) // trying to find which adapter was used
+      {
+        for (const String& a : working_adapters)
+        {
+          if (key.find(a) == 0)
+          {
+            // used adapter found
+            adapter = a;
```

Comment:
forgot putting the key of this hit into `p`?

Reply:
maybe make two separate loops?
one for finding the adapter name, the other for extracting the keys?